src/test/java/be/seeseemelk/mockbukkit/WorldMockTest.java

```diff
@@ -263,6 +264,19 @@ void spawnZombieTest()
 		assertTrue(world.getEntities().size() > 0);
 	}
 
+	@Test
+	void spawnHorseTest()
+	{
+		WorldMock world = new WorldMock();
+		Location location = new Location(world, 100, 20, 50);
+		Entity horse = world.spawnEntity(location, EntityType.HORSE);
+		assertEquals(100, horse.getLocation().getBlockX());
+		assertEquals(20, horse.getLocation().getBlockY());
+		assertEquals(50, horse.getLocation().getBlockZ());
+		assertTrue(world.getEntities().size() > 0);
+		assertInstanceOf(Horse.class, horse);
+	}
```

Comment:
I don't think the Position is necessary here

Reply:
I don't follow. You mean don't use `Location location`? Or do you mean I don't need to check the `horse.getLocation()` attributes because that is tested elsewhere?

Reply:
If you look at the other Entity Spawn Tests you see that we don't check the Position there. Thats because the Method you are using is implemented in EntityMock and also tested there. And since we don't need special Handeling of the Method in the HorseMock class, we don't need to test it here. Only write a Test for something if you implement / override it in this class.

Reply:
Got it. I removed the `.getLocation()` assertions. I assume I should still leave the rest of the test here since I did add logic to the `WorldMock` to spawn the `HorseMock`?

Reply:
Yes thats correct.